### ISSUE TITLE
[BUG] Tests timeout in BitBucket Pipeline but work locally 

### ISSUE THREAD

**Issue by tgoswami013** [microsoft/playwright#26615]
### System info
- Playwright Version: [1.37.1]
- Operating System: [BitBucket Ubuntu]
- Browser: [All]

**Description**
From last few days, every test is failing on BitBucket with timeout error. I have not changed anything in config and yml file. Also, tests are working fine locally. Later while checking recording on BrowserStack I observed that JavaScript is not loading for website, when I am running using BitBucket Pipeline. While sending commands to BrowserStack from local, JavaScript is loading.

**BitBucket YML File**

```js
pipelines:
  default:
    - step:
        name: Playwright Tests
        image: mcr.microsoft.com/playwright:v1.37.1-jammy
        script:
          - npm ci        
          - npx playwright install msedge
          - npx playwright install chrome
          - DEBUG=pw:browser* npx playwright test tests/ --config=playwright.config.ts --project='Windows Chrome'
        artifacts:
         - playwright-report/index.html
```

**Error on BitBucket**

```js
    Test timeout of 60000ms exceeded while running "beforeEach" hook.
  Error: locator.click: Page closed
    =========================== logs ===========================
    waiting for locator('a[href*=\'signin.html\']')
    ============================================================
       at ../page-objects/components/header.component.ts:44
      42 |
      43 |      async clickLoginIcn(){
    > 44 |           await this.loginIcn.click()
         |                               ^
      45 |      }
```

**Chrome Browser Debug Logs**

```js
pw:browser <launching> /opt/google/chrome/chrome --disable-field-trial-config --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-back-forward-cache --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-component-update --no-default-browser-check --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose,MediaRouter,DialMediaRouteProvider,AcceptCHFrame,AutoExpandDetailsElement,CertificateTransparencyComponentUpdater,AvoidUnnecessaryBeforeUnloadCheckSync,Translate --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --headless --hide-scrollbars --mute-audio --blink-settings=primaryHoverType=2,availableHoverTypes=2,primaryPointerType=4,availablePointerTypes=4 --no-sandbox --user-data-dir=/tmp/playwright_chromiumdev_profile-ukrtqv --remote-debugging-pipe --no-startup-window +0ms
  pw:browser <launched> pid=1928 +6ms
  pw:browser [pid=1928][err] [0822/180244.917043:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory +70ms
  pw:browser [pid=1928][err] [0822/180244.964134:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory +47ms
  pw:browser [pid=1928][err] [0822/180244.964209:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory +0ms
  pw:browser [pid=1928][err] [0822/180244.966290:ERROR:zygote_host_impl_linux.cc(273)] Failed to adjust OOM score of renderer with pid 1949: Permission denied (13) +2ms
  pw:browser [pid=1928][err] [0822/180244.968899:WARNING:bluez_dbus_manager.cc(247)] Floss manager not present, cannot set Floss enable/disable. +3ms
  pw:browser [pid=1928][err] [0822/180244.990228:ERROR:zygote_host_impl_linux.cc(273)] Failed to adjust OOM score of renderer with pid 1953: Permission denied (13) +21ms
  pw:browser [pid=1928][err] [0822/180245.032181:WARNING:sandbox_linux.cc(393)] InitializeSandbox() called with multiple threads in process gpu-process. +42ms
  pw:browser [pid=1928] <gracefully close start> +1m
  pw:browser [pid=1928] <process did exit: exitCode=0, signal=null> +33ms
  pw:browser [pid=1928] starting temporary directories cleanup +0ms
  pw:browser [pid=1928] finished temporary directories cleanup +2ms
  pw:browser [pid=1928] <gracefully close end> +0ms
  pw:browser <launching> /opt/google/chrome/chrome --disable-field-trial-config --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-back-forward-cache --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-component-update --no-default-browser-check --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose,MediaRouter,DialMediaRouteProvider,AcceptCHFrame,AutoExpandDetailsElement,CertificateTransparencyComponentUpdater,AvoidUnnecessaryBeforeUnloadCheckSync,Translate --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --headless --hide-scrollbars --mute-audio --blink-settings=primaryHoverType=2,availableHoverTypes=2,primaryPointerType=4,availablePointerTypes=4 --no-sandbox --user-data-dir=/tmp/playwright_chromiumdev_profile-MZTeqX --remote-debugging-pipe --no-startup-window +0ms
  pw:browser <launched> pid=2003 +8ms
  pw:browser [pid=2003][err] [0822/180345.577131:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory +72ms
  pw:browser [pid=2003][err] [0822/180345.581239:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory +3ms
  pw:browser [pid=2003][err] [0822/180345.581304:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory +0ms
  pw:browser [pid=2003][err] [0822/180345.583805:ERROR:zygote_host_impl_linux.cc(273)] Failed to adjust OOM score of renderer with pid 2024: Permission denied (13) +2ms
  pw:browser [pid=2003][err] [0822/180345.585834:WARNING:bluez_dbus_manager.cc(247)] Floss manager not present, cannot set Floss enable/disable. +2ms
  pw:browser [pid=2003][err] [0822/180345.609571:ERROR:zygote_host_impl_linux.cc(273)] Failed to adjust OOM score of renderer with pid 2028: Permission denied (13) +24ms
  pw:browser [pid=2003][err] [0822/180345.650429:WARNING:sandbox_linux.cc(393)] InitializeSandbox() called with multiple threads in process gpu-process. +41ms
  pw:browser [pid=2003] <gracefully close start> +1m
  pw:browser [pid=2003] <process did exit: exitCode=0, signal=null> +21ms
  pw:browser [pid=2003] starting temporary directories cleanup +0ms
  pw:browser [pid=2003] finished temporary directories cleanup +2ms
  pw:browser [pid=2003] <gracefully close end> +0ms
```

**Comment by yury-s** [comment]
Sounds like a BitBucket or BrowserStack issue since playwright hasn't changed and the issue is not reproducible locally. I see that you install msedge and chrome, could it be that they have been updated recently that caused the issue? Make sure you use the same Chrome version locally as on the CI. If you think this is a bug in Playwright, we'll need a self-contained repro that we could run locally.

**Comment by tgoswami013** [comment]
Hi Thanks, this is CDN Cache or application specific issue. Earlier we found on BitBucket, now it is detected on local browser incognito window, and with playwright local execution.

I also executed same scripts on Production environment using BitBucket Pipeline where application code is not deployed, scripts are working fine, we have not observed timeout or JavaScript issues.

We can say that this is not a Playwright, BitBucket, Browser or BrowserStack Issue. This is issue with CDN Cache or application build specific issue.

Thanks for developing this amazing tool and great support.
